completion/available/pip.completion.bash

```diff
@@ -2,9 +2,36 @@
 # Of course, you should first install the pip, say on Debian:
 # sudo apt-get install python-pip
 # sudo apt-get install python3-pip
-# If the pip package is installed within virtual environments, say, python managed by pyenv,
-# you should first initilization the corresponding environment.
-# So that the pip/pip3 is in system's path.
-if command -v pip >/dev/null; then
-  eval "$(pip completion --bash)"
+
+# If the pip package is installed within virtual environments,
+# you should first install pip for the corresponding environment.
+
+# Fix pip completion for running it within/outside of pyenv/virtualenv/venv/conda environment.
+_regex="(^|;[ ]*)_pip_completion_hook([ ]*;|$)"
+
+_pip_completion_hook() {
+  local _pip
+  # For pip resides within the pyenv/virtualenv/venv/conda environments:
+  if [ -n "$VIRTUAL_ENV" ] && [ -x "$VIRTUAL_ENV/bin/pip" ]; then
+    _pip="$VIRTUAL_ENV/bin/pip"
+  elif [ -n "$CONDA_PREFIX" ] && [ -x "$CONDA_PREFIX/bin/pip" ]; then
+    _pip="$CONDA_PREFIX/bin/pip"
+  # For pip resides outside of a virtual environment:
+  elif [ -x /usr/bin/pip ]; then
+    _pip=/usr/bin/pip
+  fi
+  if [ -n "$_pip" ]; then
+    eval "$($_pip completion --bash)"
+  fi 
+  unset _pip 
+}
+
+
+if [ -z "$PROMPT_COMMAND" ]; then
```

Comment:
Please use the existing `safe_append_prompt_command` function instead of this `if` block - it should do what you're trying to do here.

https://github.com/Bash-it/bash-it/blob/b35d41464c0bd390e573f7423eaaa63666521c70/themes/base.theme.bash#L511

Reply:
Thanks for this code snippet.